tests/Unit/FilterBookingRequestsTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Tests\Unit;
+
+use App\Models\BookingRequest;
+use Carbon\Carbon;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class FilterBookingRequestsTest extends TestCase
+{
+    use RefreshDatabase;
+    /**
+     * Test filtering booking request with status
+     *
+     * @return void
+     */
+    public function test_filter_by_status(){
+        $this->withoutExceptionHandling();
+        $user = $this->createUserWithPermissions(['bookings.approve']);
+        // Create room with boolean attribute as true and make sure it exists
+        // and is returned by the filter endpoint
+        $booking = BookingRequest::factory()->create([
+            'status' => 'review']);
+        $this->assertDatabaseHas('booking_requests', [
+            'id' => $booking->id,
+        ]);
+
+
+        $response = $this->actingAs($user)->postJson('/api/filterBookingRequests', ['status_list' => ['review' => true]]);
+        $response
+            ->assertStatus(200);
+
+        $res_id = collect($response->json())->pluck('id');
+        $this->assertContains($booking->id, $res_id);
+
+        $response2 = $this->actingAs($user)->postJson('/api/filterBookingRequests', ['status_list' => ['review' => false]]);
+        $response2
+            ->assertStatus(200);
+        $res_id_2 = collect($response2->json())->pluck('id');
```

Comment:
you should also perform a `$response->assertSessionHasNoErrors()` after each 200 check.
it helps us catch flaky tests later on

Reply:
On it.